sklearn/cluster/tests/test_bicluster.py

```diff
@@ -264,13 +239,7 @@ def test_spectralcoclustering_parameter_validation(params, type_err, err_msg):
             ValueError,
             r"Incorrect parameter n_clusters has value: \('abc', 3\)",
         ),
-        ({"method": "unknown"}, ValueError, "Unknown method: 'unknown'"),
-        ({"n_components": 0}, ValueError, "n_components == 0, must be >= 1."),
-        ({"n_components": 1.5}, TypeError, "n_components must be an instance of"),
         ({"n_components": 3, "n_best": 4}, ValueError, "n_best == 4, must be <= 3."),
-        ({"n_best": 0}, ValueError, "n_best == 0, must be >= 1."),
-        ({"n_best": 1.5}, TypeError, "n_best must be an instance of"),
-        ({"svd_method": "unknown"}, ValueError, "Unknown SVD method: 'unknown'"),
     ],
 )
 def test_spectralbiclustering_parameter_validation(params, type_err, err_msg):
```

Comment:
The other two errors
```
E           AssertionError: Regex pattern "Incorrect parameter n_clusters has value: \\('abc', 3\\)" does not match "The 'n_clusters' parameter of KMeans must be an int in the range [1, inf). Got 'abc' instead.".
```
If I understand correctly `n_clusters` is constrained from the base class but also checked individually as complex parameter: this generates two incompatible errors. I guess the error string in the parametrization should be changed accordingly.

Reply:
Thanks, I have also removed these additional checks on the n_clusters.

Reply:
However, the check on the n_components depends on the value of another param, n_best.